### PR TITLE
Provide consistent examples of service invocation for all SDKs

### LINKED ISSUE #1969: Provide consistent SDK examples for all building blocks
**Issue**: 
 Currently there are inconsistencies across Building Block SDK examples.There are a couple cases where a building block only has one or two SDK examples but there is support for that building block in other SDKs

Screen Shot 2021-11-17 at 2 47 32 PM

**The Solution**: 
Provide consistency across Building Block SDK examples

For each building block provide an SDK example (where supported) that shows how to use that building block. Each example should follow the same scenario across SDKs.

### PR DESCRIPTION
**Pull request by amulyavarote**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Modified service invocation documentation page (under building blocks) for dapr.io

## Issue reference

https://github.com/dapr/docs/issues/1969

### CODE REVIEW

**Comment by yaron2** [comment]
Can you please add an issue reference?

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md [hunk 1]
```diff
@@ -8,26 +8,133 @@ weight: 2000
 
 This article describe how to deploy services each with an unique application ID, so that other services can discover and call endpoints on them using service invocation API.
 
+## Example:
+
+There are two services called order processing service and checkout service. Dapr used in order processing service is used to invoke checkout service. Both order processing service and checkout service have dapr side cars.
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md]:
```suggestion
The below code examples loosely describe an application that processes orders. In the examples, there are two services - an order processing service and a checkout service. Both services have Dapr sidecars and the order processing service uses Dapr to invoke the checkout method in the checkout service.
```

Comment (greenie-msft) [reply daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md]: @amulyavarote - Reworded this a bit. Thoughts?

**Comment by greenie-msft** [comment]
> Can you please add an issue reference?

@yaron2 This partially fulfills: https://github.com/dapr/docs/issues/1969

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md (hunk 1 above)
Comment (amulyavarote) [reply daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md]: Looks good!

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md [hunk 2]
```diff
@@ -8,26 +8,133 @@ weight: 2000
 
 This article describe how to deploy services each with an unique application ID, so that other services can discover and call endpoints on them using service invocation API.
 
+## Example:
+
+The below code examples loosely describe an application that processes orders. In the examples, there are two services - an order processing service and a checkout service. Both services have Dapr sidecars and the order processing service uses Dapr to invoke the checkout method in the checkout service.
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md:13]:
```suggestion
The below code examples loosely describe an application architecture that processes orders. In the examples, there are two services - an order processing service and a checkout service. Each service has a Dapr sidecar and the order processing service uses Dapr to invoke the checkout method in the checkout service.
```

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md [hunk 3]
```diff
@@ -63,103 +170,173 @@ spec:
 
 {{< /tabs >}}
 
+## Step 2: Invoke the service
+
+Dapr uses a sidecar, decentralized architecture. To invoke an application using Dapr, you can use the `invoke` API on any Dapr instance.
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md]:
```suggestion
To invoke an application using Dapr, you can use the `invoke` API on any Dapr instance.
```

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md [hunk 4]
```diff
@@ -63,103 +170,173 @@ spec:
...
+
+The sidecar programming model encourages each applications to talk to its own instance of Dapr. The Dapr instances discover and communicate with one another.
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md]:
```suggestion
The sidecar programming model encourages each application to interact with its own instance of Dapr. The Dapr sidecars discover and communicate with one another.
```

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md [hunk 5]
```diff
@@ -63,103 +170,173 @@ spec:
...
+
+Below are the example code snippets leveraging Dapr SDKs.
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md]:
```suggestion
Below are code examples that leverage Dapr SDKs for service invocation.
```

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md [hunk 6]
```diff
@@ -63,103 +170,173 @@ spec:
...
+
+{{< tabs Dotnet Java Python Go Javascript curl CLI >}}
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md]: @amulyavarote is php not supported here?

File: daprdocs/content/en/operations/support/support-release-policy.md [hunk 7]
```diff
@@ -39,13 +39,14 @@ The table below shows the versions of Dapr releases that have been tested togeth
...
 | Jun 16th 2021 | 1.2.2</br> | 1.2.0 | Java 1.1.0 </br>Go 1.1.0 </br>PHP 1.1.0 </br>Python 1.1.0 </br>.NET 1.2.0 | 0.6.0 | Unsupported         |
-| Jul 26th 2021 | 1.3</br>   | 1.3.0 | Java 1.2.0 </br>Go 1.2.0 </br>PHP 1.1.0 </br>Python 1.2.0 </br>.NET 1.3.0 | 0.7.0 | Supported           |
-| Sep 14th 2021 | 1.3.1</br> | 1.3.0 | Java 1.2.0 </br>Go 1.2.0 </br>PHP 1.1.0 </br>Python 1.2.0 </br>.NET 1.3.0 | 0.7.0 | Supported           |
-| Sep 15th 2021 | 1.4</br>   | 1.4.0 | Java 1.3.0 </br>Go 1.3.0 </br>PHP 1.2.0 </br>Python 1.3.0 </br>.NET 1.4.0 | 0.8.0 | Supported           |
-| Sep 22nd 2021 | 1.4.1</br>   | 1.4.0 | Java 1.3.0 </br>Go 1.3.0 </br>PHP 1.2.0 </br>Python 1.3.0 </br>.NET 1.4.0 | 0.8.0 | Supported
-| Sep 24th 2021 | 1.4.2</br>   | 1.4.0 | Java 1.3.0 </br>Go 1.3.0 </br>PHP 1.2.0 </br>Python 1.3.0 </br>.NET 1.4.0 | 0.8.0 | Supported           |
-| Oct 7th 2021  | 1.4.3</br>   | 1.4.0 | Java 1.3.0 </br>Go 1.3.0 </br>PHP 1.2.0 </br>Python 1.3.0 </br>.NET 1.4.0 | 0.8.0 | Supported (current) |
-
+| Jul 26th 2021 | 1.3</br>   | 1.3.0 | Java 1.2.0 </br>Go 1.2.0 </br>PHP 1.1.0 </br>Python 1.2.0 </br>.NET 1.3.0 | 0.7.0 | Unsupported           |
```
Comment (greenie-msft) [inline daprdocs/content/en/operations/support/support-release-policy.md]: @amulyavarote, these files should not be included in this PR as changes

File: daprdocs/content/en/operations/support/support-release-policy.md [hunk 8]
```diff
@@ -59,15 +60,22 @@ General guidance on upgrading can be found for [self hosted mode]({{<ref self-ho
 |                          |                 1.1.2 |                    1.2.2 |
 |                          |                 1.2.2 |                    1.3.1 |
 |                          |                 1.3.1 |                    1.4.3 |
+|                          |                 1.4.3 |                    1.5.0 |
```
Comment (greenie-msft) [inline daprdocs/content/en/operations/support/support-release-policy.md]: @amulyavarote, these files should not be included in this PR as changes

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md [hunk 9]
```diff
@@ -63,103 +170,173 @@ spec:
...
+
+
+{{% codetab %}}
+```bash
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md]: if you change this to "dotnet", you will get syntax highlighting for dotnet. 

Same goes for the other languages

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md [hunk 10]
```diff
@@ -63,103 +170,173 @@ spec:
...
-if __name__ == '__main__':
-    app.run()
 ```
+{{% /codetab %}}
 
-This Python app exposes an `add()` method via the `/add` endpoint.
 
-## Step 3: Invoke the service
+{{% codetab %}}
+```bash
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md]: if you change bash to java, you will get java syntax highlighting....same goes for all the rest of the languages

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md [hunk 11]
```diff
@@ -63,103 +170,173 @@ spec:
...
+To invoke a 'GET' endpoint:
 
 ```bash
-curl http://localhost:3500/v1.0/invoke/cart/method/add -X DELETE
+curl http://localhost:3602/v1.0/invoke/checkout/method/checkout/100
 ```
 
-Dapr puts any payload returned by the called service in the HTTP response's body.
-
 ### Additional URL formats
```
Comment (greenie-msft) [inline daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md]: We should not include HTTP proxy in the curl example. That section should stand by itself as an option for customers looking to adopt Dapr coming from an application that already has http client code calling existing endpoints and they do not want to change over to using the invoke url format.

My recommendation is to leave the "Additional URL formats" section outside of the code blocks. Moving forward, we should elaborate on this feature

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md [hunk 12]
```diff
@@ -8,26 +8,133 @@ weight: 2000
...
+dapr run --app-id orderprocessing --app-port 6001 --dapr-http-port 3601 --dapr-grpc-port 60001 --app-ssl npm start
+
+```
+
+{{% /codetab %}}
+
+
 {{% codetab %}}
 
 ### Setup an ID using Kubernetes
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md]:
```suggestion
### Set an app-id when deploying to Kubernetes
```

File: daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md (hunk 6 above)
Comment (amulyavarote) [reply daprdocs/content/en/developing-applications/building-blocks/service-invocation/howto-invoke-discover-services.md]: I have php code. But I have not executed it yet. Thats the reason I have not included it in the document.

**Review by greenie-msft — APPROVED** [review]

**Review by willtsai — APPROVED** [review]
LGTM, thanks for updating the image as well to represent Dapr sidecars.
